src/System.Console/src/System/ConsolePal.WinRT.cs

```diff
@@ -0,0 +1,256 @@
+
+        internal sealed class ControlCHandlerRegistrar
+        {
+            internal void Register() { throw new PlatformNotSupportedException(); }
+
+            internal void Unregister() { throw new PlatformNotSupportedException(); }
+        }
+
+        public static Stream OpenStandardInput() { throw new PlatformNotSupportedException(); }
+
+
+        public static Stream OpenStandardOutput() { return new NoOpStream(FileAccess.Write); }
+
+        public static Stream OpenStandardError() { return new NoOpStream(FileAccess.Write); }
+
+        public static Encoding InputEncoding
+        {
+            get { return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false); }
+        }
+
+        public static Encoding OutputEncoding
+        {
+            get { return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false); }
+        }
+
+        public static bool KeyAvailable { get { throw new PlatformNotSupportedException(); } }
+
+        public static ConsoleKeyInfo ReadKey(bool intercept)
+        {
+            if (Console.IsInputRedirected)
+            {
+                // We could leverage Console.Read() here however
+                // windows fails when stdin is redirected.
+                throw new InvalidOperationException(SR.InvalidOperation_ConsoleReadKeyOnFile);
+            }
+            throw new PlatformNotSupportedException();
+        }
+
+        public static bool TreatControlCAsInput
+        {
+            get
+            {
+                if (Console.IsInputRedirected)
+                    return false;
+                throw new PlatformNotSupportedException();
+            }
+            set
+            {
+                if (!Console.IsInputRedirected)
+                    throw new InvalidOperationException(SR.InvalidOperation_ConsoleReadKeyOnFile);
+            }
+        }
+
+        public static ConsoleColor ForegroundColor
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static ConsoleColor BackgroundColor
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static void ResetColor() { throw new PlatformNotSupportedException(); }
+
+        public static bool NumberLock { get { throw new PlatformNotSupportedException(); } }
+
+        public static bool CapsLock { get { throw new PlatformNotSupportedException(); } }
+
+        public static int CursorSize
+        {
+            get { return 100; }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static string Title
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set
+            {
+                if (!Console.IsOutputRedirected)
```

Comment:
Especially since it looks like we throw a PlatformNotSupportedException from IsOutput/InputRedirected....?